spec/lib/contextual_logger_spec.rb

```diff
@@ -171,6 +172,79 @@
     end
   end
 
+  describe 'with_context without block' do
+    it 'returns a context handler' do
+      expect(@logger.with_context(service: 'test_service')).to be_a(ContextualLogger::Context::Handler)
+    end
+
+    it 'prints out the wrapper context with logging' do
+      expected_log_line = {
+        service: 'test_service',
+        message: 'this is a test',
+        severity: 'INFO',
+        timestamp: Time.now,
+        progname: nil
+      }.to_json
+
+      expect_any_instance_of(Logger::LogDevice).to receive(:write).with("#{expected_log_line}\n")
+
+      handler = @logger.with_context(service: 'test_service')
+      expect(@logger.info('this is a test')).to eq(true)
+      handler.reset!
```

Comment:
This seems like the main use case for returning the handler...maybe only return the handler only if no block is provided?